nixos/modules/services/networking/openconnect.nix

```
@@ -0,0 +1,154 @@
+{ config, lib, options, pkgs, ... }: with lib;
+let
+  cfg = config.networking.openconnect;
+  openconnect = cfg.package;
+  interfaceOptions = {
+    options = {
+      gateway = mkOption {
+        description = "Gateway server to connect to.";
+        example = "gateway.example.com";
+        type = types.str;
+      };
+
+      protocol = mkOption {
+        description = ''
+          Protocol to use. One of <literal>anyconnect</literal>,
+          <literal>array</literal>, <literal>nc</literal>,
+          <literal>pulse</literal>, <literal>gp</literal>,
+          <literal>f5</literal> and <literal>fortinet</literal>
+        '';
+        example = "anyconnect";
+        type = types.str;
```

```suggestion
        type = types.enum [ "anyconnet" "array" ... ];
```
having the type list all its valid values instead of doing it in the description is usually preferrable